packages/story-editor/src/components/panels/document/publish/publish.js

```diff
@@ -116,73 +158,115 @@ function PublishPanel() {
           featuredMedia: {
-            id: resource.id,
-            height: resource.sizes?.full?.height || resource.height,
-            url: resource.sizes?.full?.source_url || resource.src,
-            width: resource.sizes?.full?.width || resource.width,
+            id: image.id,
+            height: image.sizes?.full?.height || image.height,
+            url: image.sizes?.full?.url || image.url,
+            width: image.sizes?.full?.width || image.width,
           },
         },
       }),
     [updateStory]
   );
 
-  // Used for onSelect prop in MediaUpload component.
-  // @todo Enforce square image while selecting in Media Library.
-  const handleChangePublisherLogo = useCallback(
-    (resource) => {
-      updateStory({
-        properties: {
-          publisherLogo: {
-            id: resource.id,
-            url: resource.sizes?.full?.source_url || resource.src,
-            width: resource.sizes?.full?.width || resource.width,
-            height: resource.sizes?.full?.height || resource.height,
-          },
+  const onNewPublisherLogoSelected = ({ id, src }) => {
+    const newLogo = { id, url: src };
+    addPublisherLogo(id);
+    setPublisherLogos((logos) => [...logos, newLogo]);
+    onPublisherLogoChange(newLogo);
+  };
+
+  const onPublisherLogoChange = (option) => {
+    updateStory({
+      properties: {
+        publisherLogo: {
+          id: option.id,
+          url: option.url,
         },
-      });
-    },
-    [updateStory]
-  );
+      },
+    });
+  };
 
-  const publisherLogoErrorMessage = useMemo(() => {
-    let message = __('No file types are currently supported.', 'web-stories');
+  const getErrorMessage = (message) => {
+    let returnedMessage = __(
+      'No file types are currently supported.',
+      'web-stories'
+    );
 
     if (allowedImageFileTypes.length) {
-      message = sprintf(
-        /* translators: %s: list of allowed file types. */
-        __('Please choose only %s as publisher logo.', 'web-stories'),
+      returnedMessage = sprintf(
+        message,
         translateToExclusiveList(allowedImageFileTypes)
       );
     }
 
-    return message;
-  }, [allowedImageFileTypes]);
+    return returnedMessage;
+  };
 
-  const posterErrorMessage = useMemo(() => {
-    let message = __('No file types are currently supported.', 'web-stories');
+  const publisherLogoErrorMessage = getErrorMessage(
+    /* translators: %s: list of allowed file types. */
+    __('Please choose only %s as publisher logo.', 'web-stories')
+  );
+  const posterErrorMessage = getErrorMessage(
+    /* translators: %s: list of allowed file types. */
+    __('Please choose only %s as a poster.', 'web-stories')
+  );
 
-    if (allowedImageFileTypes.length) {
-      message = sprintf(
-        /* translators: %s: list of allowed file types. */
-        __('Please choose only %s as a poster.', 'web-stories'),
-        translateToExclusiveList(allowedImageFileTypes)
+  const openMediaPicker = useMediaPicker({
```

Comment:
⚠️ 

This needs to use the `<MediaUpload` component from the config, something like this:

```js
const { MediaUpload } = useConfig();

// ...

<MediaUpload
  onSelect={onNewPublisherLogoSelected}
  onSelectErrorMessage={publisherLogoErrorMessage}
  type={allowedImageMimeTypes}
  render={renderUploadButtonIcon}
title={__('Select as publisher logo', 'web-stories')}
/>
```